### PR TITLE
feat(dev-console): Adds feedback config page

### PR DESCRIPTION
**Pull request by deshmukhmayur**
# Closes ONEPLAT-1054

# Explain the feature/fix

Initial implementation for the Feedback Configuration page in the developer console.

## Does this PR introduce a breaking change

No.

## Screenshots

<details>
<summary>View Screenshots</summary>

Screenshot 2021-07-30 at 4 12 51 PM

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?

### CODE REVIEW

File: packages/developer-console-spa/src/components/ConfigureFeedback.tsx [hunk 1]
```diff
@@ -1,16 +1,155 @@
...
+
+  const saveFeedbackConfig = (event: any) => {
+    event.preventDefault();
+
+    if ( !canSubmit ) {
+      return;
+    }
+
+    console.log( feedbackConfig );
+    console.log( { ...app, feedback: editableFeedbackConfig } );
```
Comment (riginoommen) [inline packages/developer-console-spa/src/components/ConfigureFeedback.tsx:52]:
```suggestion
```

File: packages/developer-console-spa/src/hooks/useFeedbackConfig.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        if ( !res || !res.data ) {
+          setLoading( false );
+          return;
+        }
+        setFeedbackConfig( res.data.app?.feedback ?? {} );
+        setLoading( false );
+      } )
+      .catch( err => {
+        window.OpNotification?.danger( {
+          subject: 'There was some error fetching feedback config.',
```
Comment (riginoommen) [inline packages/developer-console-spa/src/hooks/useFeedbackConfig.ts:29]:
```suggestion
          subject: 'There was some error fetching feedback configuration.',
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/developer-console-spa/src/components/ConfigureFeedback.tsx (hunk 1 above)
Comment (deshmukhmayur) [reply packages/developer-console-spa/src/components/ConfigureFeedback.tsx]: I was still working on that part, hence the draft PR.
I've updated the commits.

File: packages/developer-console-spa/src/hooks/useFeedbackConfig.ts (hunk 2 above)
Comment (deshmukhmayur) [reply packages/developer-console-spa/src/hooks/useFeedbackConfig.ts]: Updated.

**Review by riginoommen — APPROVED** [review]

**Comment by deshmukhmayur** [comment]
This PR should not be merged until the release branch is fixed as this change is not supposed to be released yet.
Will be making it "Ready for Review" once the release branch has been recreated and the workflow refined.

File: packages/developer-console-spa/src/components/AddAppForm.tsx [hunk 3]
```diff
@@ -42,7 +42,8 @@ export default function AppAppForm () {
           window.OpNotification.success( { subject: 'App Created Succssfully!' } );
           history.push( res.data.app.appId );
         }
-      } ).catch( err => {
+      } )
+      .catch( err => {
```
Comment (ghanlohar) [inline packages/developer-console-spa/src/components/AddAppForm.tsx:46]: Add type to err

File: packages/developer-console-spa/src/components/ConfigureFeedback.tsx [hunk 4]
```diff
@@ -1,16 +1,179 @@
...
+
+  const saveFeedbackConfig = (event: any) => {
+    event.preventDefault();
+
+    if ( !canSubmit ) {
+      return;
+    }
+
+    updateFeedbackConfigService( app.id, editableFeedbackConfig )
+      .then( updatedApp => {
```
Comment (ghanlohar) [inline packages/developer-console-spa/src/components/ConfigureFeedback.tsx:60]: Add type here as well

File: packages/developer-console-spa/src/components/ConfigureFeedback.tsx [hunk 5]
```diff
@@ -1,16 +1,179 @@
...
+        setFeedbackConfig( editableFeedbackConfig );
+        window.OpNotification?.success( { subject: 'Feedback Saved!' } );
+      } )
+      .catch( err => {
```
Comment (ghanlohar) [inline packages/developer-console-spa/src/components/ConfigureFeedback.tsx:64]: Add type to the argument

File: packages/developer-console-spa/src/hooks/useFeedbackConfig.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    if ( !appId ) {
+      return;
+    }
+    const abortController = new AbortController();
+    const signal = abortController.signal;
+
+    setLoading( true );
+
+    gqlClient( { query: appFeedbackConfig, variables: { appId } }, signal )
+      .then( res => {
```
Comment (ghanlohar) [inline packages/developer-console-spa/src/hooks/useFeedbackConfig.ts:19]: Add type to `res` argument

File: packages/developer-console-spa/src/hooks/useFeedbackConfig.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        if ( !res || !res.data ) {
+          setLoading( false );
+          return;
+        }
+        setFeedbackConfig( res.data.app?.feedback ?? {} );
+        setLoading( false );
+      } )
+      .catch( err => {
```
Comment (ghanlohar) [inline packages/developer-console-spa/src/hooks/useFeedbackConfig.ts:27]: Add type to `err` argument

File: packages/developer-console-spa/src/services/feedbackConfig.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  return gqlClient( {
+    query: updateApp,
+    variables: {
+      id,
+      app: {
+        feedback
+      }
+    }
+  } )
+    .then( res => {
```
Comment (ghanlohar) [inline packages/developer-console-spa/src/services/feedbackConfig.ts:14]: Add type to `res` argument

File: packages/developer-console-spa/src/services/feedbackConfig.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+      if ( res.errors && !res?.data?.updateApp ) {
+        const errMessage = res.erors.map( ( err: any ) => err.message ).join( ', ' );
+        throw new Error( errMessage );
+      }
+      return res.data.updateApp;
+    } )
+    .catch( err => {
```
Comment (ghanlohar) [inline packages/developer-console-spa/src/services/feedbackConfig.ts:21]: Add type to `err` argument

**Review by ghanlohar — CHANGES_REQUESTED** [review]

File: packages/developer-console-spa/src/components/AddAppForm.tsx (hunk 3 above)
Comment (deshmukhmayur) [reply packages/developer-console-spa/src/components/AddAppForm.tsx:46]: Not needed, as the type is inherited from it's parent.

Comment (deshmukhmayur) [reply packages/developer-console-spa/src/components/AddAppForm.tsx:46]: Screenshot 2021-07-30 at 6 44 57 PM

File: packages/developer-console-spa/src/components/AddAppForm.tsx [hunk 10]
```diff
@@ -42,7 +42,8 @@ export default function AppAppForm () {
...
         window.OpNotification.danger( { subject: 'An error occured when deleting the App.', body: 'Please try again later.' } );
```
Comment (ghanlohar) [inline packages/developer-console-spa/src/components/AddAppForm.tsx:47]:
```suggestion
        window.OpNotification?.danger( { subject: 'An error occured when deleting the App.', body: 'Please try again later.' } );
```

**Review by ghanlohar — CHANGES_REQUESTED** [review]

File: packages/developer-console-spa/src/components/AddAppForm.tsx (hunk 10 above)
Comment (deshmukhmayur) [reply packages/developer-console-spa/src/components/AddAppForm.tsx]: Updated.

**Review by ghanlohar — APPROVED** [review]

**Pull request merged**